Sources/SwiftParser/Nominals.swift

```diff
@@ -0,0 +1,269 @@
+    arena: __shared SyntaxArena
+  ) {
+    self.init(attributes: attributes, modifiers: modifiers, unexpectedBeforeIntroducerKeyword, actorKeyword: introducerKeyword, unexpectedBeforeIdentifier, identifier: identifier, genericParameterClause: primaryOrGenerics, inheritanceClause: inheritanceClause, genericWhereClause: genericWhereClause, members: members, arena: arena)
+  }
+
+  static func parsePrimaryOrGenerics(_ parser: inout Parser) -> RawGenericParameterClauseSyntax? {
+    return parser.parseGenericParameters()
+  }
+}
+
+extension RawStructDeclSyntax: NominalTypeDeclarationTrait {
+  init(
+    attributes: RawAttributeListSyntax?,
+    modifiers: RawModifierListSyntax?,
+    _ unexpectedBeforeIntroducerKeyword: RawUnexpectedNodesSyntax?,
+    introducerKeyword: RawTokenSyntax,
+    _ unexpectedBeforeIdentifier: RawUnexpectedNodesSyntax?,
+    identifier: RawTokenSyntax,
+    primaryOrGenerics: RawGenericParameterClauseSyntax?,
+    inheritanceClause: RawTypeInheritanceClauseSyntax?,
+    genericWhereClause: RawGenericWhereClauseSyntax?,
+    members: RawMemberDeclBlockSyntax,
+    arena: __shared SyntaxArena
+  ) {
+    self.init(attributes: attributes, modifiers: modifiers, unexpectedBeforeIntroducerKeyword, structKeyword: introducerKeyword, unexpectedBeforeIdentifier, identifier: identifier, genericParameterClause: primaryOrGenerics, inheritanceClause: inheritanceClause, genericWhereClause: genericWhereClause, members: members, arena: arena)
+  }
+
+  static func parsePrimaryOrGenerics(_ parser: inout Parser) -> RawGenericParameterClauseSyntax? {
+    return parser.parseGenericParameters()
+  }
+}
+
+extension RawEnumDeclSyntax: NominalTypeDeclarationTrait {
+  init(
+    attributes: RawAttributeListSyntax?,
+    modifiers: RawModifierListSyntax?,
+    _ unexpectedBeforeIntroducerKeyword: RawUnexpectedNodesSyntax?,
+    introducerKeyword: RawTokenSyntax,
+    _ unexpectedBeforeIdentifier: RawUnexpectedNodesSyntax?,
+    identifier: RawTokenSyntax,
+    primaryOrGenerics: RawGenericParameterClauseSyntax?,
+    inheritanceClause: RawTypeInheritanceClauseSyntax?,
+    genericWhereClause: RawGenericWhereClauseSyntax?,
+    members: RawMemberDeclBlockSyntax,
+    arena: __shared SyntaxArena
+  ) {
+    self.init(attributes: attributes, modifiers: modifiers, unexpectedBeforeIntroducerKeyword, enumKeyword: introducerKeyword, unexpectedBeforeIdentifier, identifier: identifier, genericParameters: primaryOrGenerics, inheritanceClause: inheritanceClause, genericWhereClause: genericWhereClause, members: members, arena: arena)
+  }
+
+  static func parsePrimaryOrGenerics(_ parser: inout Parser) -> RawGenericParameterClauseSyntax? {
+    return parser.parseGenericParameters()
+  }
+}
+
+extension Parser {
+  /// Parse a nominal type decl declaration - class, struct, enum, or actor.
+  mutating func parseNominalTypeDeclaration<T>(
+    _ T: T.Type,
+    _ attrs: DeclAttributes,
+    _ handle: RecoveryConsumptionHandle
```

Comment:
Any reason why you decided to make these unlabeled? Also, while you’re at it, `handle` could probably be better named `introducerHandle` or `keywordHandle` (I have a mild preference for the former but don’t care too much).

Reply:
It was less a decision and more a copy paste of what we already had (with a rename):
```
 public mutating func parseClassDeclaration(
     _ attrs: DeclAttributes,
     _ handle: RecoveryConsumptionHandle
   ) -> RawClassDeclSyntax {
```

I don't have a strong preference here, how much do you care?

Reply:
If I wrote this, I would have labeled them. But if you want to merge this as-is, go ahead.

Reply:
I'm pretty sure you did add `handle` 🤔. Anyway, I went ahead and add them. Hopefully you're okay with `for` :P